src/xmpp/stanza.c

```diff
@@ -995,6 +995,9 @@ stanza_create_caps_query_element(xmpp_ctx_t* ctx)
 #endif
         }
     }
+
+    account_free(account);
```

Comment:
It should work. Ideally, in the future we should just get 1 parameter instead of fetching full account.